### PR TITLE
Update Core API documentation

### PR DESCRIPTION
**Pull request by talekhinezh**

### CODE REVIEW

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml [hunk 1]
```diff
@@ -652,12 +577,38 @@ tags:
       `/construction/finalize` in order to get a `signed_transaction`. This can then be submitted to
       `construction/submit`. You have successfully submitted a XRD transfer transaction!
```
Comment (dhedey) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:578]: I would say `an XRD transfer`, but this is super minor :)

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml [hunk 2]
```diff
@@ -760,16 +720,23 @@ tags:
 
       ## Update Balance Store
 
-      We can now start updating some balance store given that we have balance changes.
+      Now that we can compute balance chaneg we can now start updating some balance store.
```
Comment (dhedey) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:723]: change*

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml [hunk 3]
```diff
@@ -1017,11 +1044,6 @@ tags:
       the validators performing consensus. As a consequence of this, it is _not_ guaranteed to be increasing, and it
```
Comment (dhedey) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:1044]: I've realised I'm missing a closing bracket in the `post_transaction_state_accumulator` line above.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml [hunk 4]
```diff
@@ -1017,11 +1044,6 @@ tags:
...
       may be affected by Byzantine nodes.
 
-      Not every transaction will have such a `RoundData` operation, for example, due to multiple transactions being
```
Comment (dhedey) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:1020]: Why was this removed?

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml (hunk 3 above)
Comment (dhedey) [inline radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:1044]: I think you explained that the timestamp is actually the median - we should probably mention this. (It's certainly safer than the mean)

**Review by dhedey — COMMENTED** [review]
Good changes, thanks. A few minor notes.

**Review by dhedey — APPROVED** [review]

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml [hunk 5]
```diff
@@ -63,40 +49,35 @@ info:
...
+    transaction by transaction.
 
-    ![Data API Flow](https://raw.githubusercontent.com/radixdlt/radixdlt/feature/open-api/radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/documentation/data_sequence_flow.png)
+    ![Data API Flow](https://raw.githubusercontent.com/radixdlt/radixdlt/feature/update-documentation/radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/documentation/data_sequence_flow.png)
 
     # Construction API Flow
 
-    Integrators can make use of the Construction API to construct and submit transactions
-    to the network.
+    The Construction API can be used to construct and submit transactions to the network.
```
[the remark this answers is not in the document]
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:80]: I wouldn't say it's the only reason and maybe not even the main reason. The main reason is perhaps to be able to support offline transaction creation (something we don't support now but may do so in the long term). In this case, probably not worth mentioning reasoning for some of the APIs.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml (hunk 3 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:1044]: Would rather not specify how it's derived at this time as I would like to see the implementation of this change actually in the near future.

File: radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml (hunk 4 above)
Comment (talekhinezh) [reply radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/core/api.yaml:1020]: It's not really explained correctly as it implies *most* transactions will have a `RoundData` operation but no user transaction will have a `RoundData` operation. I'll refactor how this is worded
